README.md

```diff
@@ -1,2 +1,82 @@
 # velero-plugin
-HPE plugin for Velero
\ No newline at end of file
+
+HPE plugin for Velero.
+To take snapshots of HPE volumes through velero you need to install and configure the HPE Snapshotter plugin.
+
+## Installation
+
+Refer to <https://velero.io/docs/v1.1.0/get-started/> for installing velero.
+
+### 1. Create VolumeSnapshotLocation CRD for HPE Snapshotter
+
+```markdown
+
+apiVersion: velero.io/v1
+kind: VolumeSnapshotLocation
+metadata:
+  name: hpe-csp
+  namespace: velero
+spec:
+  provider: hpe.com/snapshotter
+  config: {
+       "secret-name": "nimble-secret",
+       "secret-namespace": "kube-system"
+  }
+```
+
+### 2. Create secret to communicate with HPE Nimble Storage CSP Service
+
+```markdown
+
+https://raw.githubusercontent.com/hpe-storage/csi-driver/master/examples/kubernetes/secret.yaml
+
+```
+
+### 3. Deploy the CSP service for HPE Nimble Storage in velero namespace
+
+The CSP yaml file is located at [nimble-csp.yaml](nimble-csp.yaml)
+
+### 4. Install velero HPE blockstore plugin
+
+```markdown
+
+velero plugin add hpestorage/velero-hpe-blockstore:beta --image-pull-policy Always
+
+```
+
+## Backup
+
+Everytime a velero backup is taken and include PVCs, it will also take HPE Nimble Storage snapshots of your volumes.
+
+```markdown
+
+velero backup create default-ns-hpe-backup --include-namespaces=default --snapshot-volumes --volume-snapshot-locations hpe-csp
```

Comment:
Can we use `default-ns-csp-backup` as an example instead?

Reply:
should i change it to `default-ns-hpe-csp-backup` instead?